x-pack/plugins/actions/server/builtin_action_types/lib/send_email.test.ts

```diff
@@ -153,7 +180,159 @@ describe('send_email module', () => {
+              "ensureUriAllowed": [MockFunction],
+              "getCustomHostSettings": [MockFunction],
+              "getMicrosoftGraphApiUrl": [MockFunction] {
+                "calls": Array [
+                  Array [],
+                ],
+                "results": Array [
+                  Object {
+                    "type": "return",
+                    "value": undefined,
+                  },
+                ],
+              },
+              "getProxySettings": [MockFunction],
+              "getResponseSettings": [MockFunction],
+              "getSSLSettings": [MockFunction],
+              "isActionTypeEnabled": [MockFunction],
+              "isHostnameAllowed": [MockFunction],
+              "isUriAllowed": [MockFunction],
+            },
+            "connectorId": "1",
+            "content": Object {
+              "message": "a message",
+              "subject": "a subject",
+            },
+            "hasAuth": true,
+            "routing": Object {
+              "bcc": Array [],
+              "cc": Array [
+                "bob@example.com",
+                "robert@example.com",
+              ],
+              "from": "fred@example.com",
+              "to": Array [
+                "jim@example.com",
+              ],
+            },
+            "transport": Object {
+              "clientId": "123456",
+              "clientSecret": "sdfhkdsjhfksdjfh",
+              "password": "changeme",
+              "service": "exchange_server",
+              "user": "elastic",
+            },
+          },
+        },
+        Object {
+          "context": Array [],
+          "debug": [MockFunction],
+          "error": [MockFunction],
+          "fatal": [MockFunction],
+          "get": [MockFunction],
+          "info": [MockFunction],
+          "log": [MockFunction],
+          "trace": [MockFunction],
+          "warn": [MockFunction],
+        },
+        Object {
+          "ensureActionTypeEnabled": [MockFunction],
+          "ensureHostnameAllowed": [MockFunction],
+          "ensureUriAllowed": [MockFunction],
+          "getCustomHostSettings": [MockFunction],
+          "getMicrosoftGraphApiUrl": [MockFunction] {
+            "calls": Array [
+              Array [],
+            ],
+            "results": Array [
+              Object {
+                "type": "return",
+                "value": undefined,
+              },
+            ],
+          },
+          "getProxySettings": [MockFunction],
+          "getResponseSettings": [MockFunction],
+          "getSSLSettings": [MockFunction],
+          "isActionTypeEnabled": [MockFunction],
+          "isHostnameAllowed": [MockFunction],
+          "isUriAllowed": [MockFunction],
+        },
+      ]
+    `);
+
+    expect(unsecuredSavedObjectsClient.create.mock.calls.length).toBe(1);
+  });
+
+  test('uses existing "access_token" from "connector_token" SO for authentication for email using "exchange_server" service', async () => {
```

Comment:
Can we add a test to check that `unsecuredSavedObjectsClient.update` is called when a connector token exists but has expired?